launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/CreateByLinkServlet.java

```diff
@@ -0,0 +1,147 @@
+    } else if (Strings.isNullOrEmpty(email)) {
+      throw new IllegalArgumentException("You are not logged in. Please try again.");
+    }
+
+    Key entityKey = null;
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity desiredFeedEntity;
+    try {
+      // Use key to retrieve entity from Datastore
+      entityKey = KeyFactory.stringToKey(id);
+      desiredFeedEntity = datastore.get(entityKey);
+    } catch (IllegalArgumentException e) {
+      // If entityId cannot be converted into a key
+      writeResponse(res, "Sorry, this is not a valid id.", HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    } catch (EntityNotFoundException e) {
+      // No matching entity in Datastore
+      writeResponse(res, "Your entity could not be found.", HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+
+    String xmlString = (String) desiredFeedEntity.getProperty(XML_STRING);
+
+    // Modify the xml string
+    RSS rssFeed = XML_MAPPER.readValue(xmlString, RSS.class);
+    Channel channel = rssFeed.getChannel();
+    channel.addItem(episodeTitle, episodeDescription, episodeLanguage, email, mp3Link);
+    String modifiedXmlString = RSS.toXmlString(rssFeed);
+    desiredFeedEntity.setProperty(XML_STRING, modifiedXmlString);
+    datastore.put(desiredFeedEntity);
+
+    // Return accessible link to client
+    String urlID = KeyFactory.keyToString(desiredFeedEntity.getKey()); // the key string associated with the entity, not the numeric ID.
+    String rssLink = BASE_URL + urlID;
+    res.setContentType("text/html");
+    res.getWriter().print(rssLink);
+  }
+
+  /**
+   * Display RSS feed xml string when a user clicks on an RSS feed from their "My RSS Feeds" page.
+   * @throws IOException
+   */
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
+    // Get ID passed in request
+    String id = req.getParameter(ID);
+    if (id == null) {
+      throw new IllegalArgumentException("Sorry, no matching Id was found in Datastore.");
+    }
+    Key urlId = null;
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity desiredFeedEntity;
+    
+    // Create entity that contains id from datastore
+    try {
+      urlId = KeyFactory.stringToKey(id);
+      desiredFeedEntity = datastore.get(urlId);
+
+      // Generate xml string
+      String xmlString = (String) desiredFeedEntity.getProperty(XML_STRING);
+      res.setContentType("text/xml");
+      res.getWriter().print(xmlString);
+    } catch (IllegalArgumentException e) {
+      // If entityId cannot be converted into a key
+      writeResponse(res, "Sorry, this is not a valid id.", HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    } catch (EntityNotFoundException e) {
+      // If there is no entity that matches the key
+      writeResponse(res, "Your entity could not be found.", HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+    
+  }
```

Comment:
why do we need this? doesn't FormHandlerServlet.doGet do the same thing?

Reply:
Spoke to podmates- we're going to use the FormHandler doGet() across other files